timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -818,6 +821,8 @@ def leaderboard(
                 raise_exception_if_failed=False,
                 use_cache=use_cache,
             )
+            assert isinstance(model_predictions, dict)
+            assert isinstance(pred_time_dict, dict)
```

Comment:
Maybe a slightly cleaner option would be to always return `Tuple[Dict[...], Dict[...]]` from `get_model_pred_dict` (equivalent to setting `record_pred_time=True`)? This would, however, require some modifications to the tests and all usages of `get_model_pred_dict` throughout the codebase.